core/src/main/java/org/eclipse/hono/config/ProtocolAdapterProperties.java

```diff
@@ -254,4 +243,17 @@ public void setTenantIdleTimeout(final Duration tenantIdleTimeout) {
     public final void setMapperEndpoints(final HashMap<String, MapperEndpoint> mapperEndpoints) {
         this.mapperEndpoints = Objects.requireNonNull(mapperEndpoints);
     }
+
+    /**
+     * Gets the configured mapper for the given key.
+     *
+     * @param key The key to identify the mapper
+     * @return the mapperEndpoint. If not found, will return {@code null}.
+     */
+    public MapperEndpoint getMapperEndpoint(final String key) {
+        if (!mapperEndpoints.containsKey(key)) {
+            return null;
+        }
```

Comment:
I would assume that this can be safely removed, can't it?

Reply:
If somebody configures a mapper on a device which doesn't exist we will get a keyNotFoundException if we ommit this check? Apart from the fact that you can catch that exception I rather not have an exception at all.

Reply:
If I am not mistaken then [Map.get()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#get(java.lang.Object)) returns `null` already if the map doesn't contain the given key ...

Reply:
You're right. Habit from another language :)